src/tools/perf/lib/libperf_thread.c

```diff
@@ -30,17 +30,19 @@ static ucs_status_t ucx_perf_thread_run_test(void* arg)
     ucx_perf_params_t* params       = &perf->params;
     ucs_status_t status;
 
-    /* new threads need explicit device association */
-    status = perf->send_allocator->init(perf);
-    if (status != UCS_OK) {
-        goto out;
-    }
-
-    if (perf->send_allocator != perf->recv_allocator) {
-        status = perf->recv_allocator->init(perf);
+    if (params->api == UCX_PERF_API_UCT) {
```

Comment:
In the UCT PR you deleted this flow, but here it is under the condition of `UCT` (?)